app/src/org/commcare/android/models/AsyncEntity.java

```diff
@@ -115,6 +115,9 @@ public Object getField(int i) {
                 } catch(XPathException xpe) {
                     xpe.printStackTrace();
                     data[i] = "<invalid xpath: " + xpe.getMessage() + ">";
+                } catch (XPathSyntaxException xse) {

```

Comment:
Question: We don't generally handle these here because we try to parse all expressions inbound when we install the app, allowing people to catch syntax errors (which should be detectable before runtime) early. Is it impossible to make sure the graph expressions are compiled during install to throw their errors up there?